embedded-graphics/src/mock_display.rs

```diff
@@ -288,6 +288,22 @@ where
     }
 }
 
+// impl MockDisplay<BinaryColor> {
```

Comment:
I think this shouldn't be here.

Reply:
I uncommented this and removed the other `from_points` method you commented on below to get `from_points` closer to what you were thinking. Should we try and merge this impl block with the `impl<C> MockDisplay<C>` above it? I tried before but ran into generics issues.

Reply:
> Should we try and merge this impl block with the impl<C> MockDisplay<C> above it? I tried before but ran into generics issues.

This would only work if we had some `C` parameter, like: `fn from_points(iter: I, color: C) -> Self`. But maybe replacing `from_points` by `draw_points` (suggested below) would be the most flexible option. Or keep both, if you like, where `from_points` only works with `BinaryColor` and `draw_points` with any `C`.

One thing I would like to have changed is the use of `Pixel::draw` in `from_points`. IMO this code should be independent of `Drawable`, so we don't use the same code path if something breaks, which could lead to false negatives.

To implement this I would suggest to revert what I sad about `set_pixel` in a previous PR (sorry for the back and forth) and implement the bounds and overdraw checks in that method. If the unchecked behavior is required the user can always disable the checks temporarily. But I would suggest to rename it to `draw_pixel` to make it clearer that it is not a simple setter, which should also be mentioned in the docs.

Reply:
I've pushed a new commit with some changes in it. I've moved the bounds checking to `set_pixel` (now renamed to `draw_pixel`) and added some docs.

In terms of `draw_points` and `from_points`, I've gone with just having `from_points` as that's all we need currently.